tests/scenario/test_setup_statuses.py

```diff
@@ -46,7 +46,21 @@ def patch_all(substrate, mock_cfg_path):
         yield
 
 
-def test_install(charm_type, substrate, vroot):
+@pytest.fixture()
+def mock_snap():
+    """Mock the charm's snap property so we don't access the host."""
+    with patch(
+        "charm.GrafanaAgentMachineCharm.snap", new_callable=PropertyMock
+    ) as mocked_property:
+        mock_snap = mocked_property.return_value
+        # Mock the .present property of the snap object so the start event doesn't try to configure
+        # anything
+        mock_snap.present = False
+        yield
+
+
+@patch("charm.SnapManifest._get_system_arch", return_value="amd64")
+def test_install(_mock_manifest_get_system_arch, charm_type, substrate, vroot, mock_snap):
```

Comment:
```suggestion
def test_install(_mock_manifest_get_system_arch, charm_type, substrate, vroot):
```
autouse fixtures please

Reply:
you don't want people to write a test, forget to add it, and have a test mess with their system :)

Reply:
Yeah this one I totally agree with.  I'm anti-autouse for many things, but this one I completely agree :D